manager/engine.go

```diff
@@ -170,6 +175,31 @@ func (m *VolumeManager) BackupSnapshot(snapshotName string, labels map[string]st
 		if err != nil {
 			logrus.Warnf("Failed to update volume LastBackup for %v due to cannot get backup target: %v", volumeName, err)
 		}
+
+		bks := &types.BackupStatus{}
+		for {
+			engines, err := m.ds.ListVolumeEngines(volumeName)
+			if err != nil {
+				logrus.Errorf("fail to get engines for volume %v", volumeName)
+				return
+			}
+
+			for _, e := range engines {
+				backupStatusList := e.Status.BackupStatus
+				for _, b := range backupStatusList {
+					if b.SnapshotName == snapshotName {
+						bks = b
+						break
+					}
+				}
+			}
+
+			if bks.Progress == 100 {
```

Comment:
If this backup failed, there would never be in 100. Check for BackupError for the error as well.